Adjust assemblies for orcharhino builds (#4947)
theforeman/foreman-documentation: guides/common/modules/con_provisioning-methods.adoc

:_mod-docs-content-type: CONCEPT

[id="provisioning-methods"]
= Provisioning methods

[role="_abstract"]
{Project} supports multiple ways to provision a host: image based, network based, boot disk based, and Discovery based.
The optimal host provisioning method depends on the type of host, the availability of DHCP, and if it is to be cloned from an existing VM template.

image::images/provisioning-decision-tree-orcharhino.svg[Decision tree for provisioning methods in {Project}]

Pick a provisioning method based on the type of host that you want to provision:

> .Cloud instances

Author: Should I change this to make vale happy? @jafiala 

```suggestion
Cloud instances::
```

Reviewer: Yup. Visually it should look almost the same, and it is a cleaner structure even in asciidoc.

* Choose image based host provisioning.

.Virtual machines
* If you want to clone the VM from a template/_golden image_, choose image based host provisioning.
* If your host has a static IP address, choose boot disk based host provisioning.
* If the network of your host provides DHCP, choose network based host provisioning.

.Bare metal hosts
* Choose Discovery based host provisioning.
* If your host has a static IP address, choose boot disk based host provisioning.
* If the network of your host provides DHCP, choose network based host provisioning.

Each provisioning method has a distinct set of steps to provision a host:

image::images/provisioning-workflow-overview-orcharhino.svg[Overview of provisioning workflow in {Project}]

> .Network Based Provisioning

Reviewer: Capitalization and hyphenation

```suggestion
::Network-based provisioning
```

. Depending on whether it is a virtual machine or bare metal host, {Project} either
** instructs the compute resource provider to create a virtual machine,
** or a bare metal host has to be started manually, that is by hand or through a lights out management tool.

> For bare metal hosts, the MAC address of the new host must be submitted to {Project}.

Reviewer: I think this works better structurally. We shouldn't use sentences that run on across multiple list items. And list items should match grammatically.

Not sure about the highest-level item though. Feel free to adjut.

```suggestion
. Start the system based on your deployment model:
* For a virtual machine, {Project} instructs the compute resource provider to create a virtual machine,
* For a bare-metal host, you must start it manually. That is by hand or through a lights out management tool.
For bare metal hosts, you must submit the MAC address of the new host to {Project}.
```

Author: Adjusted slightly and applied.

. The host boots over the network and automatically searches for a DHCP server, which in turn assigns an IP address and configures the network.
The DHCP server provides a PXE/iPXE file, which is a type of provisioning template.
+
[NOTE]
====

> {Team} recommends having {Project} configure and manage the DHCP server or even take over its duties through {ProjectServer} or {SmartProxyServer}.

Reviewer: We shouldn't recommend things. We should explain why this is the better way. What is the benefit?

Author: I am unsure, it's a copy of https://docs.orcharhino.com/or/docs/sources/guides/ubuntu/provisioning_hosts/introduction.html#Provisioning_Methods

AFAIK it's a recommendation to users that are unsure; using the `foreman-installer`-based approach is AFAIK the one that is tested/supported the best.

Reviewer: Same as the other recommendation - I'm OK with not following it here.

====
. The new host fetches the {client-provisioning-template-type} template from {ProjectServer} or {SmartProxyServer} through TFTP and starts the unattended installation and initial configuration of the operating system.
. The new host reports back to {Project} after finishing the installation.
This switches the status in {Project} to `installed`.

> .Boot Disk Based Provisioning

Reviewer: Hyphenation, capitalization

```suggestion
Boot-disk-based provisioning::
```

Author: Thanks, applied as "boot disk-based".

. The `boot_disk.iso` image contains the network configuration of the new host, that is it replaces the network configuration through DHCP in contrast to network based provisioning.
There are four different kinds of boot disk images:
** the _per-host_ image is the default image and does not require DHCP.
It only contains the network configuration and then points to {Project} for the iPXE template followed by the {client-provisioning-template-type} template.
** the _full-host_ image contains the network configuration, the kernel, the iPXE file, and the provisioning template.
The image is operating system specific.
** the _generic_ image is a reusable image which works for any host registered on {Project} and requires DHCP.

> ** the _subnet-generic_ image requires DHCP and chainloads the installation media from your {SmartProxy} assigned to the subnet of the host.

Reviewer: This works better as a dlist. The intro sentence is not ideal, it would be better to say why this is relevant here and now.

```suggestion
There are four different kinds of boot disk images:
Per-host image::: This is the default image and does not require DHCP.
It only contains the network configuration and then points to {Project} for the iPXE template followed by the {client-provisioning-template-type} template.
Full-host image::: This contains the network configuration, the kernel, the iPXE file, and the provisioning template.
The image is operating system specific.
Generic image::: This is a reusable image which works for any host registered on {Project} and requires DHCP.
Subnet-generic image::: This requires DHCP and chainloads the installation media from your {SmartProxy} assigned to the subnet of the host.
```

Reviewer: Or without the thises it could also work...
The default image...
An image that contains...
A reusable image...
An image that requires...

Author: Applied.

Author: I went with your original suggestion for now. OK?

.Operating system - template mapping
[options="header",cols="2,1"]
|===
|Operating system |Provisioning template

|AlmaLinux, Amazon Linux, CentOS, Oracle Linux, Red{nbsp}Hat Enterprise Linux, and Rocky Linux
|Kickstart templates

|Debian
|Preseed templates

|SUSE Linux Enterprise Server 16
|Agama templates

|SUSE Linux Enterprise Server 15 and 12
|AutoYaST templates

|Ubuntu
|Preseed Autoinstall templates
|===

.Discovery Based Provisioning
Discovery based provisioning is similar to boot disk based provisioning.
It uses a `discovery.iso` image to boot the host into a minimal Linux system, reports it to {Project}, and sends various facts including its MAC address to identify the host.
Afterwards, the host is listed on the discovered hosts page.
Discovery is an optional plugin.

. Depending on whether a DHCP server is available,
** the host either manually receives its network configuration and Discovery image and boots the `discovery.iso` image in an interactive way;

> ** or the host automatically fetches its network configuration and Discovery image from the DHCP server and boots the `discovery.iso` image in a non interactive way.

Reviewer: ```suggestion
. Depending on whether a DHCP server is available, the host receives its network configuration and Discovery image:
** The host manually receives its network configuration and Discovery image and boots the `discovery.iso` image in an interactive way.
** The host  automatically fetches its network configuration and Discovery image from the DHCP server and boots the `discovery.iso` image in a non-interactive way.
```

Author: Applied.

> . The `discovery.iso` image contains the information to which {ProjectServer} or {SmartProxyServer} the host must report to and collects facts about the host and sends it to {ProjectServer} or {SmartProxyServer}.

Reviewer: I think there was an extra word.

```suggestion
. The `discovery.iso` image contains the information to which {ProjectServer} or {SmartProxyServer} the host must report and collects facts about the host and sends it to {ProjectServer} or {SmartProxyServer}.
```

Author: Applied.

** If there are existing Discovery rules, the host automatically start a network based provisioning process based on Discovery rules and host groups.
** Else, the host appears on the list of discovered hosts.
The following process is very similar to creating a hew host.
+
There are two ways to start the installation of the operating system:
+
*** the host is rebooted and then gets the network based provisioning template by the DHCP server;

> *** or the installation is started by using `kexec` with the current network configuration.

Reviewer: ```suggestion
*** The host is rebooted and then gets the network based provisioning template by the DHCP server.
*** The installation is started by using `kexec` with the current network configuration.
```

Author: Applied.
